tests/cases/exports/export_structs_and_enums_as.slint

```diff
@@ -0,0 +1,25 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-1.1 OR LicenseRef-Slint-commercial
+
+//include_path: ../../helper_components
+import { ExportedStruct, ExportEnum } from "export_structs.slint";
+export { ExportedStruct as NamedStruct, ExportEnum as NamedEnum  }
+export component TestCase inherits Rectangle {
+    in-out property<ExportedStruct> st;
+    in-out property<ExportEnum> en: ExportEnum.Bonjour;
+}
+
+/*
+
+```cpp
+auto handle = TestCase::create();
+const TestCase &instance = *handle;
+assert(instance.get_en() == ExportEnum::Bonjour);
```

Comment:
If the enum is exported as `NamedEnum` , I expect to use it in this file as NamedEnum.  (same in Rust)
@tronical don't you think?  otherwise what's the point of giving it another name.

Reply:
I agree and was wondering about the same, but from the before/after output it looked like to me that that’s another bug and not a regression in this PR.

I assume they the export under alias (pub X as Y) would be part of a separate PR as per the issue filed.

Did I misread perhaps and this is a regression?

Reply:
Sorry, the PR description could've been better. :)

I'm looking into fixing named exports for enums and structs (#3599) which actually led me to this change to remove those duplicate identical enum and struct definitions that appear when adding "as Foo" to enum or struct exports.

With 1.2, the generated code looks like this:

```rust
mod slint_generatedTestComponent {
    use slint::private_unstable_api::re_exports as sp;
    use slint::private_unstable_api::re_exports::*;
    #[derive(Default, Copy, Clone, PartialEq, Debug)]
    pub enum r#TestEnum {
        #[default]
        r#A,
        r#B,
    }
    #[derive(Default, PartialEq, Debug, Clone)]
    pub struct r#TestStruct {
        pub r#a: i32,
        pub r#b: sp::SharedString,
    }
    #[derive(Default, Copy, Clone, PartialEq, Debug)]
    pub enum r#TestEnum {
        #[default]
        r#A,
        r#B,
    }
    #[derive(Default, PartialEq, Debug, Clone)]
    pub struct r#TestStruct {
        pub r#a: i32,
        pub r#b: sp::SharedString,
    }
    // ...
}
```